metrics/metrics.go

```diff
@@ -87,7 +89,19 @@ func (s *Scope) Timing(name string, value time.Duration, tags ...Tags) {
 	mergedTags := s.mergeTags(tags...).StringSlice()
 	s.c.logger.Debug("Metrics timing %s=%v %v", name, value, mergedTags)
 
-	if err := s.c.client.Timing(name, value, mergedTags, 1); err != nil {
+	var err error
+	if s.c.config.DatadogDistributions {
+		// Datadog recommends that, as distributions are a new distinct metric,
+		// they belong to a new metric name. We handle this by just slamming
+		// .distribution to end of all metrics that we submit this way
+		if !strings.HasSuffix(name, ".distribution") {
+			name = name + ".distribution"
+		}
+		err = s.c.client.Distribution(name, float64(value.Milliseconds()), mergedTags, 1)
+	} else {
+		err = s.c.client.Timing(name, value, mergedTags, 1)
+	}
```

Comment:
also just worth pointing out that I don't really golang in my day job -- not sure if there's a more appropriate idiomatic way to do the error handling in this block

Reply:
this is pretty canonical from what I understand